code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -490,6 +491,29 @@
 				if(G.summoner)
 					G.summoner << "<span class='danger'><B>[AM] has crossed your surveillance trap at [get_area(snare_loc)].</span></B>"
 
+/mob/living/simple_animal/hostile/guardian/ranged/Shoot(atom/targeted_atom)
+	var/obj/item/projectile/guardian/shard = ..()
+	if(!shard)
+		return
+	shard.color = guardiancolor
+
+/proc/techcolor2hex(var/colour)
```

Comment:
In this case colour is a word that's defined in guardian. It's what's passed in. This is similar to color2hex but while this is more snowflakey it means the colours match the guardians better, e.g. "red" would be 0xFF0000 while the red guardian might be more 0xFF2211